src/main/java/org/opensearch/searchrelevance/indices/SearchRelevanceIndicesManager.java

```diff
@@ -68,17 +68,34 @@ public SearchRelevanceIndicesManager(@NonNull ClusterService clusterService, @No
     }
 
     /**
-     * Create a search relevance index if not exists
-     * @param index - index to be created
+     * Create a search relevance index if not exists, or update mapping if index exists but has older schema version.
+     * @param index - index to be created or updated
      * @param stepListener - step listener
      */
     public void createIndexIfAbsent(final SearchRelevanceIndices index, final StepListener<Void> stepListener) {
         String indexName = index.getIndexName();
         String mapping = index.getMapping();
 
         if (clusterService.state().metadata().hasIndex(indexName)) {
-            log.debug("Index [{}] already exists, skipping creation", indexName);
-            stepListener.onResponse(null);
+            // Index exists - check if we need to update the mapping
+            try {
+                int existingVersion = getExistingSchemaVersion(indexName);
+                int currentVersion = index.getSchemaVersion();
+
+                if (existingVersion >= currentVersion) {
+                    log.debug("Index [{}] already exists with schema version [{}], skipping update", indexName, existingVersion);
+                    stepListener.onResponse(null);
+                    return;
+                }
+
+                // Existing version is older - update mapping synchronously
+                log.info("Updating index [{}] mapping from schema version [{}] to [{}]", indexName, existingVersion, currentVersion);
+                updateMappingSync(index);
```

Comment:
if we fail to update, what are the consequences, will SRW experiments work for customer? I think this may be a serious problem, we may need to retry few times (typical number of retries is 3). Then depending on the consequences we may need to fail fast.